providers/gcloud/gcloudProvider.go

```diff
@@ -111,17 +112,56 @@ func New(cfg map[string]string, metadata json.RawMessage) (providers.DNSServiceP
 		oldRRsMap:     map[string]map[key]*gdns.ResourceRecordSet{},
 		zoneNameMap:   map[string]string{},
 	}
+	if len(metadata) != 0 {
+		err := json.Unmarshal(metadata, g)
+		if err != nil {
+			return nil, err
+		}
+		if len(g.Visibility) != 0 {
+			visibilityCheck := regexp.MustCompile("^(public|private)$")
+			if ok := visibilityCheck.MatchString(g.Visibility); !ok {
+				return nil, fmt.Errorf("GCLOUD :visibility set but does not match ^(public|private)$")
```

Comment:
Maybe better to say `GCLOUD :visibility set but not one of public or private"`

Reply:
updated message